btrfs_tree_hc.go

```diff
@@ -0,0 +1,521 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package btrfs
+
+// This code was auto-generated; DO NOT EDIT!
+
+type treeKeyType uint32
+
+type objectID uint64
+
+type fileType int
+
+type fileExtentType int
+
+type devReplaceItemState int
+
+type blockGroup uint64
+
+// SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note (see headers.go)
```

Comment:
Would love to have someone check/verify if this is sufficient (see my other comment)

Reply:
Yes, this is the correct designation, and along the lines with what @AkihiroSuda said, this is just more explicit now that the "header" is in our codebase. We were always linking against GPLv2 code (libbtrfs and the requisite headers) and were already depending on this exception, which is stated as:
```
This exception is used together with one of the above SPDX-Licenses
  to mark user space API (uapi) header files so they can be included
  into non GPL compliant user space application code.
```